src/rp2_common/hardware_clocks/include/hardware/clocks.h

```diff
@@ -89,6 +89,46 @@ extern "C" {
 #define KHZ 1000
 #define MHZ 1000000
 
+/// \tag::pll_settings[]
+// There are two PLLs in RP2040:
+// The 'SYS PLL' generates the 125MHz system clock.
+// The 'USB PLL' generates the 48MHz USB clock.
+// The defines below are correct for the above two frequencies and a 12MHz crystal
+// frequency, defined as `XOSC_KHZ`.
+//
+// If you override `XOSC_KHZ`, the settings below will need to be checked and possibly
+// revised.  If you override `SYS_CLK_KHZ`, the settings below will need revised!
+// Use `vcocalc.py` to check and calculate new values if you change any of these frequencies.
+/// \end::pll_settings[]
+
+#ifndef PLL_COMMON_REFDIV
+// Software requires that the same 'reference divider' setting is used for both PLLs (although each has its own register).
+#define PLL_COMMON_REFDIV                   1
+#endif
+
+// NOTE:  PLL settings for a USB clock of 48MHz  are based on being able to get an
+// exact 1200MHz VCO, they may need revising with a different reference frequency.
+#ifndef PLL_USB_VCO_FREQ_KHZ
```

Comment:
we should #ifdef setting these defaults with USB lock being 48Mhz

Reply:
bump

Reply:
"these" defaults for VCO, DIV1, DIV2 only make sense when USB clock is 48Mhz, so #ifdef setting them based on USB clock being 48Mhz